cpp/modmesh/buffer/SimpleArray.hpp

```diff
@@ -160,7 +161,16 @@ class SimpleArrayMixinCalculators
 
     value_type sum() const
     {
-        value_type initial = 0;
+        value_type initial;
+        if constexpr (is_complex_v<value_type>)
+        {
+            initial = value_type();
+        }
+        else
+        {
+            initial = 0;
+        }
```

Comment:
When the element type is `complex`, different initialization should be performed here.